src/Glpi/Controller/VisibilityController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use Glpi\Controller\GenericFormController;
+use Glpi\Event;
+use Html;
+use Symfony\Component\HttpFoundation\InputBag;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class VisibilityController extends GenericFormController
+{
+    public function __invoke(Request $request): Response
+    {
+        if ($request->request->has('addvisibility')) {
+            return $this->addVisibility($request);
+        }
+
+        return parent::__invoke($request);
+    }
+
+    public function addVisibility(Request $request): RedirectResponse
+    {
+        $class = $request->attributes->get('class');
```

Comment:
We probably need to check here that the user is allowed to update the visibility of the item.